spec/Section 3 -- Type System.md

```diff
@@ -2113,3 +2119,97 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+### @defer
+
+```graphql
+directive @defer(
+  label: String
+  if: Boolean! = true
+) on FRAGMENT_SPREAD | INLINE_FRAGMENT
+```
+
+The `@defer` directive may be provided for fragment spreads and inline fragments
+to inform the executor to delay the execution of the current fragment to
+indicate deprioritization of the current fragment. A query with `@defer`
+directive will cause the request to potentially return multiple responses, where
+non-deferred data is delivered in the initial response and data deferred is
+delivered in a subsequent response. `@include` and `@skip` take precedence over
+`@defer`.
+
+```graphql example
+query myQuery($shouldDefer: Boolean) {
+  user {
+    name
+    ...someFragment @defer(label: "someLabel", if: $shouldDefer)
+  }
+}
+fragment someFragment on User {
+  id
+  profile_picture {
+    uri
+  }
+}
+```
+
+#### @defer Arguments
+
+- `if: Boolean! = true` - When `true`, fragment _should_ be deferred. When
+  `false`, fragment will not be deferred and data will be included in the
+  initial response. Defaults to `true` when omitted or null.
```

Comment:
it cannot be null

Reply:
I suggest to rephrase a bit this and the entire 'defer' concept. Instead of using 'should' be deferred - and then explaining that not necessarily, server can return all immediately; 
Instead, what about treating \@defer as PERMISSION to server to defer some data if it takes more time to fetch it. The client and server 'know' that getting some data MIGHT take more time; client wants to have other 'fast' data to be delivered ASAP to start rendering the content early, to improve UI experience; if some bottom grid would be empty for some time - that's OK. So the client tells the server - this data is OK to defer if it is not available immediately. Just don't wait for it to send all other non-deferred data quickly.